### PR TITLE
Hosting Configuration: Redirect user if insufficient privileges

### PR DESCRIPTION
**Pull request by danielbachhuber**
Related https://github.com/Automattic/dotcom-forge/issues/4708

## Proposed Changes

Redirects the user to My Home if they don't have sufficient privileges to visit Hosting Configuration.

## Testing Instructions

1. Create a new Business site.
2. With the site owner, visit `/hosting-config/<site>` and verify the page loads as expected.
3. Add a second user to the site as an author.
4. With the second user, visit `/hosting-config/<site>` and verify you're redirected to My Home.

### CODE REVIEW

**Review by jeroenpf — APPROVED** [review]
The new middleware looks good and works as expected:

1. A user with the manage_options privilege can see the hosting configuration page and is not redirected.
2. A user that does not have the manage_hosting privilege is redirected to the homepage of the current site.

File: client/my-sites/hosting/controller.js [hunk 1]
```diff
@@ -30,6 +31,22 @@ function waitForState( context ) {
...
+	const { store } = context;
+	await waitForState( context );
+	const state = store.getState();
+	const site = getSelectedSite( state );
+	const canManageOptions = canCurrentUser( state, site.ID, 'manage_options' );
+
+	// if site loaded, but user cannot manage site, redirect
+	if ( site && ! canManageOptions ) {
+		page.redirect( `/home/${ site.slug }` );
+		return;
```
Comment (wojtekn) [inline client/my-sites/hosting/controller.js]: Nitpick: you could shorten it and `return page.redirect()`.

File: client/my-sites/hosting/controller.js [hunk 2]
```diff
@@ -30,6 +31,22 @@ function waitForState( context ) {
 	} );
 }
 
+export async function redirectIfInsufficientPrivileges( context, next ) {
```
Comment (wojtekn) [inline client/my-sites/hosting/controller.js]: What do you think about following the current naming convention patterns e.g. `redirectIfCant*` or just `redirectIfNoAccess`?

**Review by wojtekn — APPROVED** [review]
The code change looks good and works as expected - I'm redirected to `/home/`.

File: client/my-sites/hosting/controller.js (hunk 2 above)
Comment (danielbachhuber) [reply client/my-sites/hosting/controller.js]: I will use the refactor in https://github.com/Automattic/wp-calypso/pull/85914/commits/ebf7e623a57ec556b668f4886e171189bf88cdfb once it's merged.

**Pull request merged**